### PR TITLE
`uchar.h` additions for foot

### PR DESCRIPTION
**Pull request by no92**

### CODE REVIEW

File: options/ansi/generic/uchar.cpp
Comment (ikbenlike) [inline options/ansi/generic/uchar.cpp:7]: afaik these arguments should all be marked `restrict`

File: options/ansi/include/string.h
Comment (ikbenlike) [inline options/ansi/include/string.h:54]: should this not instead check if the POSIX option is enabled?

Comment (no92) [reply options/ansi/include/string.h:54]: Yeah this should also check if the the header is enabled, but the feature test macros are correct here.

**Comment by no92** [comment]
Addressed the comments raised and rebased onto current `master`.

File: options/ansi/generic/uchar.cpp
Comment (Geertiebear) [inline options/ansi/generic/uchar.cpp:7]: Would be nice to add a test for this.

File: options/ansi/include/string.h
Comment (Dennisbonke) [inline options/ansi/include/string.h:54]: Not sure if we want to do this, discussion about it in the past were not in favor. I can't merge this before this is solved. Korona needs address this.

**Review by Dennisbonke — CHANGES_REQUESTED** [review]
One issue, and the tests.

File: options/ansi/include/string.h
Comment (no92) [reply options/ansi/include/string.h:54]: Programs can define or not define the feature test macros, which switches whether this is being included here or not.

Comment (Geertiebear) [reply options/ansi/include/string.h:54]: If downstream programs require this I'm fine with it. It's a little ugly but doesn't seem like a hill to die on. I'm not aware of the discussion Dennis is referring to, so if someone can find an argument against it from that conversation I'm all ears.

Comment (Dennisbonke) [reply options/ansi/include/string.h:54]: [Here](https://discord.com/channels/678608576151027731/678613590646849576/805403505258266654) until the end of that day for the first real discussion I found. Korona preferred patching over including strings.h with the feature test macro. If @avdgrinten signs off on this then I'm happy to merge it and dump a bunch of patches, but without his approval I can't sign off on this (unless I read the discussion in discord in a different way, then I just ask what I've been doing the past two years).

Comment (ElectrodeYT) [reply options/ansi/include/string.h:54]: I'm personally in favor of merging this; maintaining patches for ports which would otherwise not need any patches at all is not sustainable in the long run and is also an absolute waste of time. We can not possible break anything by auto-including strings.h in certain situations, and its unlikely to massively affect compile time either.

Sure, this may not be completely standard, but standards are not always followed, and this is such a minute detail with big effects that not merging this because "it's not the standard" when many apps believe it to be the standard is unproductive.

**Review by Dennisbonke — APPROVED** [review]
LGTM

**Pull request merged**
